src/Gateway/Context/ContextGatewayCommand.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\App\SDK\Gateway\Context;
+
+abstract class ContextGatewayCommand implements \JsonSerializable
+{
+    /**
+     * @var array<string, mixed>
+     */
+    public array $payload = [];
+
+    public string $keyName;
```

Comment:
can't this be readonly? once written, e.g. in the constructor of a specific gateway command, it shouldn't change right?

Reply:
It is protected now
you  can not make a property readonly, as children can not set this outside from the parents context